### PR TITLE
Optimize computation of intersection area

### PR DESCRIPTION
**Pull request by e-sha**
This commit simplifies computation of bounding box intersection area. It uses torch.clamp instead of torch.max and memory allocation

### CODE REVIEW

**Comment by e-sha** [comment]
Sory, the second commit does not correspond to the original problem. I wonder why it was added to the original pull request.

**Comment by ayooshkathuria** [comment]
Can't seem to get what the refactoring part is doing. Just getting rid of stray whitespace? Or anything particular you could point me towards?

**Comment by e-sha** [comment]
You are right. Refactoring just remove stray whitespaces. It does not correspond to the pull request and I don't understand why it was automatically added to it.

I'd like to detach the last commit from this pull request, but I don't understand how to do it.

**Comment by e-sha** [comment]
I've found a way to remove the second commit from pull request.
